python-package/lightgbm/basic.py

```diff
@@ -3516,6 +3530,31 @@ def refit(self, data, label, decay_rate=0.9, **kwargs):
         decay_rate : float, optional (default=0.9)
             Decay rate of refit,
             will use ``leaf_output = decay_rate * old_leaf_output + (1.0 - decay_rate) * new_leaf_output`` to refit trees.
+        reference : Dataset or None, optional (default=None)
+            reference for ``data``.
+        weight : list, numpy 1-D array, pandas Series or None, optional (default=None)
+            Weight for each ``data`` instance. Weight should be non-negative values because the Hessian
+            value multiplied by weight is supposed to be non-negative.
```

Comment:
Would you like to prepare another PR that will add this note in all other places where `weight` is used?

Reply:
yes, once this PR is done. Reference: Issue #4921

Reply:
Thank you very much for documenting this!